src/test_report.py

```diff
@@ -96,32 +97,89 @@ def send_mail(self, email_msg, email_server):
         else:
             self._logger.log_message(logging.INFO, "Email Sent.")
 
+    def get_test_analysis(self, nodes):
+        total_runs = len(nodes)
+        total_failures = sum(node['status'] == "fail" for node in nodes)
+        total_timeouts = sum(node['status'] == "timeout" for node in nodes)
+        return total_runs, total_failures, total_timeouts
+
+    def create_email_summary(self, root):
```

Comment:
Instead of using this function, we can add formatting to the jinja template directly.
Something like below:
```
{{'%-15s %s %s'|format("test", "|", "result")}}
{{'%s%s%s'|format("-"*16, "+", "-"*7)}}
{%- for test in tests %}
{{'%-15s %s %s'|format(test.name, "|", test.status)}}
{%- endfor %}
```
Then why would we need this `create_email_summary`?

Reply:
I was using this, but after checking what we have in the legacy back-end I thought could be a good idea create a summary. It sounds more flexible and easy to scale, because creating a summary we can also do the proper formatting. Does it make sense?

Reply:
In one of my previous PRs, I did the same thing and used this function for formatting. But as per @gctucker, we are replacing the old backend with API, hence we need to avoid using the older implementation. On the other hand, I find template syntax a lot simpler than this function.

Reply:
> > kernelci-pipeline-test_report | test                      | result | build                    | kind
> > kernelci-pipeline-test_report | --------------------------+--------+--------------------------+-----
> > kernelci-pipeline-test_report | staging-stable-20220620.0 | pass   | 62b0bcccefebc707f60d78c9 | node
> > kernelci-pipeline-test_report | staging-stable-20220620.0 | fail   | 62b0bcccefebc707f60d78c9 | node
> > kernelci-pipeline-test_report | staging-stable-20220620.0 | pass   | 62b0bcccefebc707f60d78c9 | node
> > kernelci-pipeline-test_report | staging-stable-20220620.0 | pass   | 62b0bcccefebc707f60d78c9 | node
> 
> There's a problem here: test should be something like `check-describe`, `staging-stable-20220620.0` is the git describe which is already in the summary at the top of the report, `62b0bcccefebc707f60d78c9` is the git sha1 which is also the same for all rows and in the summary, and we don't have kernel builds at the moment. Then I don't think the `kind` column is too helpful.
> 
> We should probably just have `test` with the test name e.g. `check-describe` and `result` with pass / fail / other.


updated..

```
kernelci-pipeline-test_report | MIME-Version: 1.0
kernelci-pipeline-test_report | To: kernelci-results-staging@groups.io
kernelci-pipeline-test_report | From: bot@kernelci.org
kernelci-pipeline-test_report | Subject: testing/master v5.18-rc4-31-g2a987e65025e: 5 runs 5 fails (check-describe)
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | --===============3708184138874473614==
kernelci-pipeline-test_report | Content-Type: text/plain; charset="utf-8"
kernelci-pipeline-test_report | MIME-Version: 1.0
kernelci-pipeline-test_report | Content-Transfer-Encoding: quopri
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | testing/master v5.18-rc4-31-g2a987e65025e: 5 runs 5 fails (check-describe)
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | Tests Summary
kernelci-pipeline-test_report | -------------
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | Build details:
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | Tree:     testing
kernelci-pipeline-test_report | Branch:   master
kernelci-pipeline-test_report | Describe: v5.18-rc4-31-g2a987e65025e
kernelci-pipeline-test_report | URL:      https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git
kernelci-pipeline-test_report | SHA1:     babf0bb978e3c9fce6c4eba6b744c8754fd43d8e
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | Revision details:
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | test           | result
kernelci-pipeline-test_report | ---------------+-------
kernelci-pipeline-test_report | check-describe | fail  
kernelci-pipeline-test_report | check-describe | fail  
kernelci-pipeline-test_report | check-describe | fail  
kernelci-pipeline-test_report | check-describe | fail  
kernelci-pipeline-test_report | check-describe | fail  
kernelci-pipeline-test_report | --===============3708184138874473614==--
kernelci-pipeline-test_report | 
kernelci-pipeline-test_report | 06/22/2022 03:40:18 PM UTC [INFO] Server Connected.
kernelci-pipeline-test_report | 06/22/2022 03:40:19 PM UTC [INFO] Email Sent.

```

Reply:
> In one of my previous PRs, I did the same thing and used this function for formatting. But as per @gctucker, we are replacing the old backend with API, hence we need to avoid using the older implementation. On the other hand, I find template syntax a lot simpler than this function.

ok, I see, thanks for sharing this information :) I will update the code.